### PR TITLE
Feature/enable code splitting

### PR DESCRIPTION
**Pull request by MarkiyanPyts**
Enabled code splitting support 

I removed strict error, because new rule sourceType: "module" enables ES modules syntax and in ES modules strict is jus not needed (in fact eslint errors our when discovers  "use strict" making us to remove them)

It would be fine for FD js but since we are introducing autoformat on file save it's not good for DS BD js.
Sometimes in BD js strict is needed.

### CODE REVIEW

File: config/.eslintrc.js [hunk 1]
```diff
@@ -8,6 +8,12 @@ module.exports = {
         jquery: true
     },
     globals: {},
+    parser: "babel-eslint",
```
Comment (danechitoaie) [inline config/.eslintrc.js]: Don't we need to also install this? Add as a dependency in package.json?

File: config/.eslintrc.js [hunk 2]
```diff
@@ -8,6 +8,12 @@ module.exports = {
...
+    parserOptions: {
+        sourceType: "module",
```
Comment (danechitoaie) [inline config/.eslintrc.js]: Will this still allow for previous common js modules to work? As in is it backwards compatible?

Comment (MarkiyanPyts) [reply config/.eslintrc.js]: Yes only difference is that it was complaining about "use strict" in modules now which I disabled. also it improves ES6 support

File: config/.eslintrc.js (hunk 1 above)
Comment (danechitoaie) [reply config/.eslintrc.js]: @MarkiyanPyts what about this?

Comment (MarkiyanPyts) [reply config/.eslintrc.js]: yes but it needs to be dependency in project package JSON…
though if you think that we can add it to OSF linter as dev dependency we can

**Pull request merged**